v2/tools/generator/internal/jsonast/swagger_type_extractor.go

```diff
@@ -569,19 +629,40 @@ func (extractor *SwaggerTypeExtractor) expandEnumsInPath(
 		}
 
 		enum, isEnum := paramType.(*astmodel.EnumType)
-		if isEnum {
-			toReplace := fmt.Sprintf("{%s}", resolved.Name)
-			var expanded []string
-
-			for _, path := range results {
-				for _, opt := range enum.Options() {
-					theOption := strings.Trim(opt.Value, "\"")
-					expanded = append(expanded, strings.ReplaceAll(path, toReplace, theOption))
-				}
-			}
+		if !isEnum {
+			// We only expand enums, skip everything else
+			continue
+		}
+
+		if len(enum.Options()) == 0 {
+			// This shouldn't happen but guard against it anyway
+			continue
+		}
+
+		// resource types come on the even counts: Microsoft.Group/resourceType/{resourceName}/{childResourceTypeParameter}/{childName}
+		isResourceTypeParameter := i%2 == 0
+		isDefaultEnum := false
+		if len(enum.Options()) == 1 {
+			opt := enum.Options()[0]
+			theOption := strings.Trim(opt.Value, "\"")
```

Comment:
Are double quotes (`"`) always used or are other forms sometimes used? I'm thinking primarily of single quotes (`'`) but also of more exotic kinds.

Reply:
For strings, yes, `"` is always used. That's an artifact of how we store enums in the code generator, not how they are specified in Swagger.

They are stored that way because they become string literals (which have "'s around them).

There are some enum types (ints, etc) where the value isn't a string literal and so doesn't have quotes, but those values won't be equal to "default" (the only enum we expand) so this comparison is safe.